invenio_rdm_records/services/services.py

```diff
@@ -338,8 +338,8 @@ def get_provider(self, scheme, provider_name=None, client_name=None):
                 return provider_class(client)
             elif provider_config["system_managed"]:
                 # use as default the client configured for the provider
-                client_name = provider_class.name
-                client = self.get_client(client_name)
+                provider_name = provider_class.name
```

Comment:
minor: it was `client_name` because the provider name was the default client name... but I think is more understandable as you  changed it now :)